sdk/core/core/inc/az_http_transport.h

```diff
@@ -89,24 +89,76 @@ az_http_request_get_header(_az_http_request const* request, int32_t index, az_pa
  *   `az_http_request_get_parts(request, &method, NULL, NULL)`
  *
- * This function is expected to be used by transport layer only.
+ * @remarks This function is expected to be used by transport layer only.
  *
- * @param request HTTP request to get parts from.
+ * @param[in] request HTTP request to get parts from.
  * @param[out] out_method __[nullable]__ Pointer to write HTTP method to. Use `NULL` to ignore
  * getting this value.
  * @param[out] out_url __[nullable]__ Pointer to write URL to. Use `NULL` to ignore getting this
  * value.
  * @param[out] out_body __[nullable]__ Pointer to write HTTP request body to. Use `NULL` to ignore
  * getting this value.
  *
- * @retval AZ_OK Success.
+ * @retval An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
  */
 AZ_NODISCARD az_result az_http_request_get_parts(
     _az_http_request const* request,
     az_http_method* out_method,
     az_span* out_url,
     az_span* out_body);
 
+/**
+ * @brief This function is expected to be used by transport adapters like curl. Use it to write
+ * content from \p write to \p response.
+ *
+ * @remarks This is a convenient way of hiding the internal implementation of az_htt_response.
+ *
+ * @remarks Parameter \p write can be an empty span. If so, nothing will be written.
+ *
+ * @param[in] response Pointer to an az_http_response.
+ * @param[in] write This is an az_span with the content to be written into response.
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
+ *         - #AZ_ERROR_INSUFFICIENT_SPAN_SIZE if the \p response is not big enough to contain the
+ * \p write content
+ */
+AZ_NODISCARD az_result az_http_response_write_span(az_http_response* response, az_span write);
+
+/**
+ * @brief This function is expected to be used by transport adapters like curl. Use it to write
+ * just one single byte to an az_http_response.
+ *
+ * @remarks This is a convenient way of hiding the internal implementation of az_htt_response.
+ *
+ * @remarks Parameter \p write can be an empty span. If so, nothing will be written.
+ *
+ * @param[in] response Pointer to an az_http_response.
+ * @param[in] byte This is a single byte to be written into response.
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful.
+ *         - #AZ_ERROR_INSUFFICIENT_SPAN_SIZE if the \p response is not big enough to contain one
+ * extra byte.
+ */
+AZ_NODISCARD az_result az_http_response_write_u8(az_http_response* response, uint8_t byte);
+
+/**
+ * @brief Returns the count of headers on the request
+ *        Each header is an az_pair
+ *
+ */
+AZ_NODISCARD int32_t _az_http_request_headers_count(_az_http_request const* request);
+
+/**
+ * @brief This is the general signature that any transport adapter like curl needs to implement.
+ *
+ * @param[in] p_request Points to an az_http_request
```

Comment:
Docs like these aren't super useful and redundant. It is clear what the parameter is from reading the code.


```suggestion
 * @param[in] p_request Points to an #az_http_request.
```